tests/functional/commands/test_insert_command.py

```diff
@@ -144,3 +144,31 @@ def test_insert_two_hyperparameters(database, monkeypatch):
     assert trial['status'] == 'new'
     assert trial['params'][0]['value'] == 1
     assert trial['params'][1]['value'] == 2
+
+
+@pytest.mark.usefixtures("clean_db")
+def test_insert_with_version(database, monkeypatch, script_path):
+    """Try to insert a single trial inside a specific version"""
+    monkeypatch.chdir(os.path.dirname(os.path.abspath(__file__)))
+    orion.core.cli.main(["init_only", "-n", "experiment",
+                         "-c", "./orion_config_random.yaml", script_path, "-x~normal(0,1)"])
+    orion.core.cli.main(["init_only", "-n", "experiment",
+                         "-c", "./orion_config_random.yaml", script_path, "-x~normal(0,1)",
+                         "-y~+normal(0,1)"])
+
+    orion.core.cli.main(["insert", "-n", "experiment", "--version", "1",
+                         "-c", "./orion_config_random.yaml", script_path, "-x=1"])
+
+    exp = list(database.experiments.find({"name": "experiment", "version": 1}))
```

Comment:
From now on we should avoid using the database backend directly in the tests (unless testing the backend itself) and rather use the StorageProtocol interface.